ts/nni_manager/training_service/reusable/trialDispatcher.ts

```diff
@@ -279,64 +296,113 @@ class TrialDispatcher implements TrainingService {
                     this.log.info(`TrialDispatcher: GPU scheduler is enabled.`)
                     this.enableGpuScheduler = true;
                 }
-                this.runnerSettings.enableGpuCollector = this.enableGpuScheduler;
 
-                this.runnerSettings.command = this.trialConfig.command;
                 // Validate to make sure codeDir doesn't have too many files
                 await validateCodeDir(this.trialConfig.codeDir);
                 break;
+            case TrialConfigMetadataKey.PLATFORM_LIST: {
+                const platforms: string[] = value.split(",");
+                for(const platform of platforms) {
+                    let environmentService: EnvironmentService;
+                    switch(platform) {
+                        case 'local':
+                            environmentService = new LocalEnvironmentService();
+                            break;
+                        case 'remote':
+                            environmentService = new RemoteEnvironmentService();
+                            break;
+                        case 'aml':
+                            environmentService = new AMLEnvironmentService();
+                            break;
+                        case 'pai':
+                            environmentService = new OpenPaiEnvironmentService();
+                            break;
+                        default:
+                            throw new Error(`${platform} not supported!`);
+                    }
+                    if (!this.commandChannelDict.has(environmentService.getCommandChannelName)) {
+                        switch(environmentService.getCommandChannelName) {
```

Comment:
This is not necessary, channel should be from service.

Reply:
updated.